util/types.c

```diff
@@ -50,13 +51,21 @@ char *uint128_t_to_string_tg(nvme_uint128_t val, bool thousands_grouping)
 	static char str[60];
 	int idx = 60;
 	__u64 div, rem;
+	char *sep = localeconv()->thousands_sep;
+	int len = sep ? strlen(sep) : 0;
```

Comment:
If I got this right, the separator is always one byte. So if `sep != NULL` than `len == 1`.

Reply:
About the struct lconv char * pointer members it is described as below so looks `len == 0` is possible if `sep != NULL` since `""` that is `len == 0` but not `len == 1` is set to the thousands_sep member.
https://www.open-std.org/jtc1/sc22/wg14/www/docs/n1570.pdf
```
7.11.2 Numeric formatting convention inquiry
7.11.2.1 The localecon
...
Description
3 The members of the structure with type char * are pointers to strings, any of which
(except decimal_point) can point to "", to indicate that the value is not available in
the current locale or is of zero length.
```

Reply:
Argh, so the spec allows "" as a valid separator, which is for obvious reasons pretty stupid. Thanks for looking this up.